### ISSUE TITLE
Instance argument is printed as explicit argument

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#4857]
Consider the following example

```agda
postulate
  A   : Set
  myA : A

module ModuleB {{a : A}} where

  record RecordB : Set₁ where
    field
      fieldB : Set

  open RecordB {{...}} public

open ModuleB {{myA}}

postulate
  myRecB : RecordB

record RecordC : Set where
  field
    fieldC  : fieldB {{myRecB}}

test : RecordC
RecordC.fieldC test = {!!}
```
When asked for the type of the goal, Agda prints `fieldB myRecB`. However, this is not well-typed: `fieldB` has type `{{r : RecordB}} → Set`. This seems to be a printing bug only, since Agda does accept arguments of the correct type `fieldB {{myRecB}}`.

**Comment by UlfNorell** [comment]
It's a display form problem. The issue seems to be the fix of #1932 (420bd20f50) which translates `λ x → x .f` into `f` to enable display forms for projections. Unfortunately it ignores the hiding on the lambda also turning `λ {{x}} → x .f` into `f`.

**Comment by UlfNorell** [comment]
This turned into quite the rabbit hole. The code that generated module application display forms was broken and completely impenetrable. PR to fix the problem and make things more sane.
